core/src/wallet/bitcoin/synchronizers/BlockchainExplorerAccountSynchronizer.cpp

```diff
@@ -95,5 +135,191 @@ namespace ledger {
         std::shared_ptr<api::ExecutionContext> BlockchainExplorerAccountSynchronizer::getSynchronizerContext() {
             return getContext();
         }
+
+        int BlockchainExplorerAccountSynchronizer::putTransaction(soci::session &sql,
+                                                                  const BitcoinLikeBlockchainExplorerTransaction &transaction,
+                                                                  const std::shared_ptr<CommonBuddy> &buddy) {
+            // In case the transaction is in mempool, keep in memory before inserting it in database.
+            if (transaction.block.isEmpty()) {
+                std::static_pointer_cast<BitcoinSynchronizationBuddy>(buddy)->mempoolTransaction.emplace_back(
+                        transaction);
+                auto keychain = buddy->account->getKeychain();
+                int flag = 0;
+                // Mark inputs
+                for (auto& input : transaction.inputs) {
+                    if (input.address.hasValue()) {
+                        keychain->markAsUsed(input.address.getValue());
+                        flag |= BitcoinLikeAccount::FLAG_TRANSACTION_CREATED_SENDING_OPERATION;
+                    }
+                }
+                // Mark output
+                for (auto& output : transaction.outputs) {
+                    if (output.address.hasValue()) {
+                        keychain->markAsUsed(output.address.getValue());
+                        flag |= BitcoinLikeAccount::FLAG_TRANSACTION_CREATED_SENDING_OPERATION;
+                    }
+                }
+                return flag;
```

Comment:
We can avoid redundant code by doing something like :

```suggestion    
    auto const markAddress = [keychain](auto &txIo) {
        auto flag = 0;
       
        for (auto &io : txIo) {
            if (io.address.hasValue()) {
                keychain->markAsUsed(io.address.getValue());
                flag |= BitcoinLikeAccount::FLAG_TRANSACTION_CREATED_SENDING_OPERATION;
            }
        }

        return flag;
    };

    return markAddress(transaction.inputs) | markAddress(transactoin.outputs);
```